internal/controller/controllers/bmh_agent_controller.go

```diff
@@ -917,6 +920,23 @@ func (r *BMACReconciler) reconcileBMH(ctx context.Context, log logrus.FieldLogge
 		return reconcileComplete{stop: true}
 	}
 
+	if r.PauseProvisionedBMHs {
+		// Add 'status' and 'paused' annotations for provisioned BMHs
+		if bmh.Status.Provisioning.State == bmh_v1alpha1.StateProvisioned {
+			result := r.addBMHStatusAndPausedAnnotations(log, bmh)
+			if res := r.handleReconcileResult(ctx, log, result, bmh); res != nil {
+				return res
+			}
+		}
+		// Remove 'paused' annotation if BMH's provisioning state is missing
+		if bmh.Status.Provisioning.State == bmh_v1alpha1.StateNone {
```

Comment:
Why only in this state? Would it be safer to add it for `Provisioned` and remove it in any other case?

Reply:
What about when we unbind a host as well? In that case don't we reattach a provisioned BMH and reprovision it?

Same for delete.

Reply:
> What about when we unbind a host as well? In that case don't we reattach a provisioned BMH and reprovision it?
> 
> Same for delete.

So I've tried the late-binding flow, i.e. using the 'cluster-reference' annotation: https://github.com/openshift/assisted-service/blob/0de391170144d918651fba707bce5c984520bffe/internal/controller/controllers/bmh_agent_controller.go#L558
Worth a unit-test for it, will add.

Reply:
I'm saying I'd expect something to need to be added/changed in [here](https://github.com/openshift/assisted-service/blob/bb582c87450dae59cf50a6be9f2bcd10e4c293f7/internal/controller/controllers/bmh_agent_controller.go#L773-L824) to deal with removing the paused annotation like we do for detached.

Reply:
Oh, right. Good catch, need a similar handling probably. Thanks:)